### DISCUSSION TITLE
Paste from clipboard after screenshot not working

### DISCUSSION THREAD

**Discussion by ghost** [flxzt/rnote#177] [Q&A]
I saw that https://github.com/flxzt/rnote/issues/17 landed, I've updated to 5.3. The importing from clipboard works for me when/if I have an image that was copied through a file manager, in my case it was nautilus, but when taking a screenshot through [Gnome's screenshot](https://gitlab.gnome.org/GNOME/gnome-screenshot) I'm not able to paste into rnote right away, I have to go to the folder and copy it manually, if I want to paste from clipboard.

Am I missing something or is this behaviour expected?

Thanks!

**Comment by ghost** [comment]
So, I went ahead and build the project to see it for myself, and I got this:
```
 DEBUG rnote::appwindow::appwindowactions > failed to paste clipboard as vector image, unsupported mime-type
```

I'm not a rust nor gnome expert, but I'll try to find out what it is.

**Reply by flxzt** [reply]
I can tell you, its not implemented yet. ^^ Currently only SVG's and file clipboard is supported.

**Reply by ghost** [reply]
@flxzt So these are the mime_types that results from ContentFormats from clipboard:
` Mimes: GdkTexture GdkPixbuf image/png`

Well, as you've said, not implemented yet, that includes `image/png`. :)

Shouldn't `image/png` be included in that list (and maybe change from if/else statements)? Or is there something to beware of? Again, not a Rust nor Gtk expert. xD

**Reply by flxzt** [reply]
I had some trouble with the implementation when I initially wrote the clipboard import, so i set it aside. But it should definitely be revisited, because as you mentioned the usecase: "pasting from a screenshot" is a very common one.
